Be smarter with existing network names

cjeanner opened the pull request:
Until now, it was more a "all or none" politic regarding the prefix
added to the network names.

This patch allows to be slightly smarter by allowing to point to actual
network names we don't want to prefix.

This is especially interesting for multi-hypervisor layout, with an OCP
cluster deployed: since ocpbm and ocppr are managed by the 3rd party
product "dev-scripts", this new feature will allow to deploy the same
networks on remote nodes, ensuring consistency in the names all the way
around.

As a pull request owner and reviewers, I checked that:
- [X] Appropriate testing is done and actually running
- [X] Appropriate documentation exists and/or is up-to-date:
  - [X] README in the role

cjeanner commented:
/test all

lewisdenny commented:
Not sure if it's an issue @cjeanner but this is sharing commits with https://github.com/openstack-k8s-operators/ci-framework/pull/998

cjeanner commented:
> Not sure if it's an issue @cjeanner but this is sharing commits with #998

that's why I marked it as "draft" - I rebased that patch onto #998 on purpose for my local tests. Now it's all good, we're up-to-date without any trailing patch since I rebased against main after the 998 merge :)

lewisdenny commented:
recheck

lewisdenny commented:
```
quay.io/openstack-k8s-operators/rabbitmq-cluster-operator-bundle@sha256:2a09973faa91ae6d3913d61cc0ae7e124240c27f886563de87a2fbdd3aec6ad0...\\nError: initializing source docker://quay.io/openstack-k8s-operators/rabbitmq-cluster-operator-bundle@sha256:2a09973faa91ae6d3913d61cc0ae7e124240c27f886563de87a2fbdd3aec6ad0: can't talk to a V1 container registry\\n. exit status 125\"", "Error: error pulling image: Trying to pull quay.io/openstack-k8s-operators/rabbitmq-cluster-operator-bundle@sha256:2a09973faa91ae6d3913d61cc0ae7e124240c27f886563de87a2fbdd3aec6ad0...", "Error: initializing source docker://quay.io/openstack-k8s-operators/rabbitmq-cluster-operator-bundle@sha256:2a09973faa91ae6d3913d61cc0ae7e124240c27f886563de87a2fbdd3aec6ad0: can't talk to a V1 container registry", ". exit status 125",
```

lewisdenny commented:
recheck

pablintino commented:
/approve

psathyan reviewed (approved):
/lgtm
